PayPal/src/main/java/com/braintreepayments/api/PayPalInternalClient.java

```diff
@@ -68,19 +68,27 @@ public void onResult(String responseBody, Exception httpError) {
                                                 String redirectUrl = paypalPaymentResource.getRedirectUrl();
                                                 if (redirectUrl != null) {
                                                     Uri parsedRedirectUri = Uri.parse(redirectUrl);
-
-                                                    String pairingIdKey = isBillingAgreement ? "ba_token" : "token";
-                                                    String pairingId = parsedRedirectUri.getQueryParameter(pairingIdKey);
-                                                    String clientMetadataId = payPalRequest.getRiskCorrelationId() != null
-                                                            ? payPalRequest.getRiskCorrelationId() : payPalDataCollector.getClientMetadataId(context, configuration, payPalRequest.hasUserLocationConsent());
+                                                    String pairingId = findPairingId(parsedRedirectUri);
+
+                                                    String clientMetadataId = payPalRequest.getRiskCorrelationId();
+                                                    if (clientMetadataId == null) {
+                                                        PayPalDataCollectorInternalRequest dataCollectorRequest =
+                                                                new PayPalDataCollectorInternalRequest(payPalRequest.hasUserLocationConsent())
+                                                                        .setApplicationGuid(payPalDataCollector.getPayPalInstallationGUID(context));
+
+                                                        if (pairingId != null) {
+                                                            dataCollectorRequest.setRiskCorrelationId(pairingId);
+                                                        }
+                                                        clientMetadataId = payPalDataCollector.getClientMetadataId(context, dataCollectorRequest, configuration);
+                                                    }
 
                                                     if (pairingId != null) {
-                                                        payPalResponse
-                                                                .pairingId(pairingId)
-                                                                .clientMetadataId(clientMetadataId);
+                                                        payPalResponse.pairingId(pairingId);
```

Comment:
This change resolves some testing errors. Logically it seems odd to set two fields in the `PayPalResponse` based on one value being `null`.

Reply:
What changed here?
I see the chained call being separated into another chained call. Is the order of calling things important here?

Reply:
Shouldn't be the chained calls are only setters. The main difference now is `clientMetadataId` is always set.